neuralpp/experiments/experimental_inference/exact_belief_propagation.py

```diff
@@ -92,9 +93,13 @@ def expand_partial_tree_and_recompute(self, expansion_root):
         potential_expansion = self.expansion[expansion_root]
         if potential_expansion is None:
             return
-        self.expansion.expand_partial_tree_and_recompute(expansion_root)
-        self.recompute_and_propagate_result_to_ancestors(potential_expansion.node)
+        expand_to_node = potential_expansion.node
+        parent = self.full_tree.parent(expand_to_node)
+
+        self.tree.add_edge(parent, expand_to_node)
+        self.expansion.update_value(expand_to_node)
```

Comment:
As mentioned in the meeting, wouldn't be better if `PartialTreeComputation` simply invalidates a node and its parent's values when an edge is edged, but waited until a value is actually requested to compute the new values?